### ISSUE TITLE
BRASIL with mpmath function

### ISSUE THREAD

**Issue by robertodr** [c-f-h/baryrat#3]
Hello and thanks for making this package available! I get an error trying to use BRASIL with mpmath. This is my sample code:
```python
from baryrat import brasil
import mpmath as mp

mp.dps = 150

r = brasil(lambda x: mp.sqrt(x), interval=[0.0, 1.0], deg=(7,7))
```
and the error:
```
TypeError: cannot create mpf from array([0.00273905, 0.02447174, 0.0669873 , 0.12842759, 0.20610737,
       0.29663168, 0.39604415, 0.5       , 0.60395585, 0.70336832,
       0.79389263, 0.87157241, 0.9330127 , 0.97552826, 0.99726095])
```
Is this not possible? Or am I doing something wrong?

**Comment by c-f-h** [comment]
Hi,

Although many functions do, the BRASIL function itself doesn't currently support mpmath. I might look into supporting that, but it might be very slow in any case because BRASIL requires so many iterations.

For the sqrt, I generally just used double precision BRASIL and then computed the poles or whatever I might need using mpmath, which gave quite accurate results. Of course, if you really require 150 digits, then this won't work.

The other thing is that we're currently working on another algorithm which performs much better for extended precision, but it's not quite ready.

It would be interesting to know your use case, in particular how many digits you actually need!

**Comment by c-f-h** [comment]
I've published a new version 2.0, available using pip, which supports mpmath somewhat more generally.

Try this in the new version:

```python
import baryrat
from mpmath import mp, mpf
import numpy as np

mp.dps = 150

r = baryrat.brasil(np.vectorize(mp.sqrt), interval=[mpf(0), mpf(1)], deg=(7,7))
```

Notes:
* The interval must be specified in extended precision to enable this.
* The function to be approximated must be able to operate on arrays of mpf values, and `np.vectorize()` is a convenient wrapper for this.
* Your original code set `mpmath.dps = 150`, which is actually the wrong place to set the number of digits, it should be set in the `mpmath.mp` context.

Let me know if this fixes your issue!

**Comment by c-f-h** [comment]
baryrat now uses the new and much faster `flamp` package for extended precision calculations.

Furthermore the new `brane` algorithm has been added, which can compute best rational approximations to very high accuracy.

Therefore I consider this issue closed.
